apps/rebar/test/rebar_completion_SUITE.erl

```diff
@@ -0,0 +1,68 @@
+-module(rebar_completion_SUITE).
+
+-compile([export_all, nowarn_export_all]).
+
+-include_lib("common_test/include/ct.hrl").
+
+suite() ->
+    [].
+
+all() ->
+    [test_competion_gen, check_bash].
+
+groups() ->
+    [].
+
+init_per_suite(Config) ->
+    Shells = [bash],
+    ComplFile = compl_file(Config),
+    ok = filelib:ensure_dir(ComplFile),
+    [{compl_file, ComplFile}, {shells, Shells} | Config].
+
+end_per_suite(_Config) ->
+    ok.
+
+init_per_testcase(_, Config) ->
+    rebar_test_utils:init_rebar_state(Config, "completion_").
+
+end_per_testcase(_, _Config) ->
+    ok.
+
+%% test cases
+
+test_competion_gen(Config) ->
+    Shells = ?config(shells, Config),
+    ComplFile = ?config(compl_file, Config),
+    lists:foreach(fun(Shell) ->
+                    file:delete(ComplFile),
+                    completion_gen(Config, #{shell=>Shell, file=>ComplFile}),
+                    {Shell, true} = {Shell,filelib:is_file(ComplFile)}
+                  end,
+                  Shells).
+
+check_bash(Config) ->
+    ComplFile = ?config(compl_file, Config),
+    Aliases = ["rebar", "r3"],
+    Opts = #{shell => bash,
+             file => ComplFile,
+             aliases => Aliases},
+    completion_gen(Config, Opts),
+    {ok, Completion} = file:read_file(ComplFile),
+    %% function definition
+    {match, _} = re:run(Completion, "_rebar3\\(\\)\\{"),
+    %% aliases
+    CompleteCmd = "complete -o nospace -F _rebar3 ",
+    lists:foreach(fun(Alias) ->
+                    {Alias, {match, _}} = {Alias, re:run(Completion, CompleteCmd++Alias++"\n")}
+                  end,
+                  ["rebar3" | Aliases]).
```

Comment:
the one thing we may need to do with this test is make it conditional to running in environments that support bash. See this test suite for an example: https://github.com/erlang/rebar3/blob/8207d82465a1661642320d63a7bcf95b187413cf/apps/rebar/test/rebar_file_utils_SUITE.erl#L76-L85

Reply:
I thought I handled that, but now I see I didn't... I suggest using `os:find_executable/1` where argument is `atom_to_list(Shell)` (cuz we may have shells supported on win, but not on unix and mentioned case would cover all cases).

Reply:
Yeah that would work fine.